### PR TITLE
Fix issues related to react native nohoisting

### PR DESCRIPTION
**Pull request by vikr01**
- Don't use both `jest.config.js` and the `jest` key in `package.json`
- `nohoist` everything starting with `react-native-`

There's now some new issue about a `.graphql` file not being found.

### CODE REVIEW

File: packages/app/package.json [hunk 1]
```diff
@@ -32,10 +32,10 @@
     "react-devtools": "^3.4.2",
     "react-test-renderer": "16.5.2",
     "relay-compiler": "1.6.2",
-    "relay-devtools": "^1.4.0"
-  },
-  "jest": {
-    "preset": "react-native"
+    "relay-devtools": "^1.4.0",
+    "graphql-query-test-mock": "^0.9.0",
+    "nock": "^10.0.1",
```
Comment (jgcmarins) [inline packages/app/package.json]: why do we need this?

Comment (vikr01) [reply packages/app/package.json]: @sibelius

Comment (sibelius) [reply packages/app/package.json]: this is needed for graphql-query-test-mock

we should migrate to https://github.com/1stdibs/relay-mock-network-layer soon, it sounds more scalable and easier

Comment (jgcmarins) [reply packages/app/package.json]: nice

**Pull request merged**